cmd/main.go

```diff
@@ -28,6 +29,12 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed to listen: %v", err)
 	}
+	db, err := repository.Factory("memory")
```

Comment:
I assume the `Database` interface (or `db` instance here) would be passed to subsequent services, right? Allowing you to pass a mock database to those services during testing.

Reply:
correct